bundles/http/Citrix/0b4d1a6b-30f3-4134-a354-42fa008b757e/f3b61af4-efd0-43c2-9a08-a467fc8242c1.js

```diff
@@ -0,0 +1,1029 @@
+                    <QueryDateSpan>${start} - ${end}</QueryDateSpan>
+                    <Employee>
+                        <PersonIdentity personNumber="${pno.personNumber}"/>
+                    </Employee>
+                </EmployeeRequestMgmt>
+            </REQUEST>
+        </Kronos_WFC>`
+        let {response,result} = await request(body)
+        let path = result.Kronos_WFC.Response[0].EmployeeRequestMgmt[0].RequestItems[0].EmployeeGlobalTimeOffRequestItem ?? [];
+        let reqTimeOffData = path.map(value =>{
+            let com = value.Comments?.[0].Comment[0] ?? []
+            return {
+                "id":value.$.Id,
+                "statusName":value.$.StatusName,
+                "requestFor":value.$.RequestFor,
+                "startDate":value.TimeOffPeriods != undefined ? moment(value.TimeOffPeriods[0].TimeOffPeriod[0].$.StartDate,`${integrationParameters.dateFormat}`).format('yyyy-MM-DD') : '',
+                "length":value.TimeOffPeriods != undefined ? Number(value.TimeOffPeriods[0].TimeOffPeriod[0].$.Length) : '',
+                "startTime":value.TimeOffPeriods?.[0].TimeOffPeriod[0].$.StartTime ?? '',
+                "duration":value.TimeOffPeriods?.[0].TimeOffPeriod[0].$.Duration ?? '',
+                "endDate":value.TimeOffPeriods != undefined ? moment(value.TimeOffPeriods[0].TimeOffPeriod[0].$.EndDate,`${integrationParameters.dateFormat}`).format('yyyy-MM-DD') : '',
+                "personNumber":pno.personNumber,
+                "payCodeName":value.TimeOffPeriods?.[0].TimeOffPeriod[0].$.PayCodeName ?? '',
+                "CommentText":com.$?.CommentText ?? null,
+                "CommentDataSourceClientName":com.$ !=undefined ?com.Notes[0].Note[0].DataSource[0].DataSource[0].$.ClientName:null,
+                "CommentDataSourceFunctionalAreaCode":com.$ !=undefined ?Number(com.Notes[0].Note[0].DataSource[0].DataSource[0].$.FunctionalAreaCode):null,
+                "CommentDataSourceFunctionalAreaName":com.$ !=undefined ?com.Notes[0].Note[0].DataSource[0].DataSource[0].$.FunctionalAreaName:null,
+                "CommentDataSourceServerName":com.$ !=undefined ?com.Notes[0].Note[0].DataSource[0].DataSource[0].$.ServerName:null,
+                "CommentDataSourceUserName":com.$ !=undefined ?com.Notes[0].Note[0].DataSource[0].DataSource[0].$.UserName:null,
+                "CommentNotesText":com.$ !=undefined ?com.Notes[0].Note[0].$.Text:null,
+                "CommentNotesTimestamp":com.$ !=undefined ?com.Notes[0].Note[0].$.Timestamp:null,
+                "CreationDateTime":moment(value.$.CreationDateTime, `${integrationParameters.dateFormat} hh:mm A`).toDate(),
+            }    
+        })
+        dataStore.save("timeOff", reqTimeOffData);
+    }
+}
+
+async function syncRecordTimestamp(dataStore, client, session, people,integrationParameters) {
+    const start = moment().subtract(1,'d').format(integrationParameters.dateFormat)
+    const end = moment().format(integrationParameters.dateFormat)
```

Comment:
Do you always sync just last day? In the microapp the empty table says "No Punch Record Available for this week".

Reply:
PD Team is not interested to have week data in the cache and table. So we do sync for only past two days data. We will make the changes in the UI accordingly as ("No Punch Record Available for last two days").